pylint/checkers/variables.py

```diff
@@ -2400,15 +2400,8 @@ def _check_unpacking(self, inferred, node, targets):
             return
 
         # Attempt to check unpacking is properly balanced
-        values: Optional[List] = None
-        if isinstance(inferred, (nodes.Tuple, nodes.List)):
-            values = inferred.itered()
-        elif isinstance(inferred, astroid.Instance) and any(
-            ancestor.qname() == "typing.NamedTuple" for ancestor in inferred.ancestors()
-        ):
-            values = [i for i in inferred.values() if isinstance(i, nodes.AssignName)]
-
-        if values:
+        values = self._get_values_to_unpack(inferred)
+        if values is not None:
```

Comment:
Can't we return an empty list and then just compare the lengths? Thereby removing this `if`.

Reply:
It would be nice to remove the indented block but it changes the logic to do that.

Returning None indicates that the right-hand-side of the assignment is not something that can be unpacked.
So, if something like `a, b = 1` is encountered we want it to go to the `elif not utils.is_iterable(inferred):` block to trigger `unpacking-non-sequence`.

Reply:
Ah my bad, should have spotted that!